x/rollapp/keeper/rollapp.go

```diff
@@ -116,18 +117,36 @@ func (k Keeper) SetRollapp(ctx sdk.Context, rollapp types.Rollapp) {
 	), []byte(rollapp.RollappId))
 }
 
-func (k Keeper) SealRollapp(ctx sdk.Context, rollappId string) error {
+// IsRollappSealed returns true if the rollapp's genesis info is sealed
+func (k Keeper) IsGenesisSealed(ctx sdk.Context, rollappId string) bool {
+	rollapp, found := k.GetRollapp(ctx, rollappId)
+	if !found {
+		return false
+	}
+	return rollapp.GenesisInfo.Sealed
+}
+
+func (k Keeper) SetRollappAsStarted(ctx sdk.Context, rollappId string) error {
 	rollapp, found := k.GetRollapp(ctx, rollappId)
 	if !found {
 		return gerrc.ErrNotFound
 	}
 
+	if !rollapp.AllImmutableFieldsAreSet() {
+		return errorsmod.Wrap(gerrc.ErrFailedPrecondition, "seal with immutable fields not set")
+	}
+
 	if rollapp.Launched {
 		return errorsmod.Wrap(gerrc.ErrFailedPrecondition, "rollapp already launched")
 	}
 
-	if !rollapp.AllImmutableFieldsAreSet() {
-		return errorsmod.Wrap(gerrc.ErrFailedPrecondition, "seal with immutable fields not set")
+	// if genesis info is not sealed, seal it if possible
+	if !rollapp.GenesisInfo.Sealed {
```

Comment:
imo this can be checked as part of `AllImmutableFieldsAreSet` vs having 2 seperate places for checking pre-conditions or alternatively simply call `Seal` genesis which checks the precondition and fails otherwise.

Reply:
refactored a bit